src/blop/bayesian/models.py

```diff
@@ -51,17 +54,24 @@ def construct_single_task_model(
         ),
     )
 
-    input_transform = botorch.models.transforms.input.Normalize(d=X.shape[-1])
+    input_transform = botorch.models.transforms.input.Normalize(d=X.shape[-1] + 1)
```

Comment:
From what I understand about MultiTaskGPs, you should not be normalizing the task indices since they represent categorical features. To be fair, I am not aware if BoTorch internally treats the `task_feature` column as special in this regard.

I would think that you shouldn't need the extra dim here but please let me know otherwise.